examples/light-switch-app/silabs/src/AppTask.cpp

```diff
@@ -80,9 +80,98 @@ CHIP_ERROR AppTask::AppInit()
         appError(err);
     }
 
+    longPressTimer = new Timer(LONG_PRESS_TIMEOUT, OnLongPressTimeout, this);
+
     return err;
 }
 
+void AppTask::Timer::Start()
+{
+    // Starts or restarts the function timer
+    if (osTimerStart(mHandler, pdMS_TO_TICKS(LONG_PRESS_TIMEOUT)) != osOK)
+    {
+        SILABS_LOG("Timer start() failed");
+        appError(CHIP_ERROR_INTERNAL);
```

Comment:
will this crash/never return? Otherwise mIsActive should not fall through to be set to true as start failed.

Reply:
appError crashes the app.